### PR TITLE
Docs: Add Bitcoin Solidity Helper Library Documentation

### PR DESCRIPTION
**Pull request by owans**
## What
- Added Bitcoin Transactions Solidity Helper Library Documentation

## Why
- Developer library docs

## Refs
- [Task](https://rsklabs.atlassian.net/browse/DV-380?atlOrigin=eyJpIjoiYTAxNzMxY2EzYzY1NGFhZDhkZTFlOWJlZTk4MTVjNjciLCJwIjoiaiJ9)

### CODE REVIEW

File: docs/02-developers/08-libraries/bitcoin-tx-solidity-helper.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+## Features of the Library
+
+The features of the Bitcoin Solidity Helper library include:
+1. Bitcoin transaction output parsing: This accurately extracts and organizes transaction outputs from raw Bitcoin transactions. It is able to receive a raw tx and return an array of structures with the tx outputs.
+2. Bitcoin transaction hashing: This calculates the cryptographic hash of a Bitcoin transaction, ensuring its authenticity and integrity. It receives a raw tx and returns its hash.
+3. Bitcoin transaction output script validation: This verifies the validity and type of output scripts within a Bitcoin transaction, allowing for specific data extraction. It receives a raw output script, validates that it is from a specific type and returns a result. E.g. receive a raw null-data script and return the embedded data in it
+4. Bitcoin address generation: is able to generate Bitcoin the address from a specific script and also to validate if a given address was generated from a script or not.
+Bitcoin address validation: This checks if a Bitcoin address conforms to a particular type or format. It validates if a Bitcoin address is of a given type or not.
```
Comment (Luisfc68) [inline docs/02-developers/08-libraries/bitcoin-tx-solidity-helper.md]:
```suggestion
5. Bitcoin address validation: This checks if a Bitcoin address conforms to a particular type or format. It validates if a Bitcoin address is of a given type or not.
```

File: docs/02-developers/08-libraries/bitcoin-tx-solidity-helper.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+```solidity
+BtcUtils.TxRawOutput[] memory outputs = BtcUtils.getOutputs(btcTx);
+```
+
+Then the user is able to perform any validation:
+```solidity
+require(expectedValue <= outputs[0].value, "incorrect amount");
+```
+
+:::info[Important]
```
Comment (Luisfc68) [inline docs/02-developers/08-libraries/bitcoin-tx-solidity-helper.md]: I can't see the enhanced version of this in the PR preview (not sure if is an issue or not)

**Review by Luisfc68 — APPROVED** [review]
lgtm

File: docs/02-developers/08-libraries/bitcoin-tx-solidity-helper.md (hunk 1 above)
Comment (owans) [reply docs/02-developers/08-libraries/bitcoin-tx-solidity-helper.md]: Noted. Thanks

File: docs/02-developers/08-libraries/bitcoin-tx-solidity-helper.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+This function is specifically useful to interact with the [rootstock native bridge](/concepts/powpeg/), as many of its functions have a transaction id as parameter. For example, by using the transaction hash function, it is easy to know how many confirmations a Bitcoin block has inside a smart contract function.
+
+### Example code with explanation
+Based on the example stated in the previous section, after validating that a specific transaction has an output paying a certain amount to an address. We need to know if that transaction has enough confirmations:
+
+Here's an example:
+
+```solidity
+BtcUtils.TxRawOutput[] memory outputs = BtcUtils.getOutputs(btcTx);
+require(expectedValue <= outputs[0].value, "incorrect amount");
```
Comment (owans) [inline docs/02-developers/08-libraries/bitcoin-tx-solidity-helper.md:118]: Are you referring to this explanation @Luisfc68 ?

Comment (Luisfc68) [reply docs/02-developers/08-libraries/bitcoin-tx-solidity-helper.md:118]: no, to the "info" part, but I don't know if formatting this is supported by the PR preview so don't worry

image

Comment (owans) [reply docs/02-developers/08-libraries/bitcoin-tx-solidity-helper.md:118]: Ok. Cool

**Review by ivegabr — APPROVED** [review]
ok

**Pull request merged**